finality-aleph/src/block/mod.rs

```diff
@@ -160,14 +160,20 @@ pub enum ChainStatusNotification<H: Header> {
 /// A stream of notifications about the chain status in the database changing.
 /// We assume that this will return all the events, otherwise we will end up with a broken state.
 #[async_trait::async_trait]
-pub trait ChainStatusNotifier<H: Header> {
+pub trait ChainStatusNotifier<H: Header>: Send {
     type Error: Debug + Display;
 
     /// Returns a chain status notification when it is available.
     /// This method's implementation must be cancellation safe.
     async fn next(&mut self) -> Result<ChainStatusNotification<H>, Self::Error>;
 }
 
+pub trait BlockchainEvents<H: Header>: Send {
```

Comment:
It's required, because we have to `Clone` `ChainStatusNotifier`s, each time when starting a new session in `NodeSessionManagerImpl`. It seems that it should be possible to avoid cloning there, but because of the `Runnable` abstraction it is not straightforward.